test/parallel/test-node-output.mjs

```diff
@@ -0,0 +1,134 @@
+    { name: 'message/async_error_microtask_main.js' },
+    { name: 'message/async_error_nexttick_main.js' },
+    { name: 'message/async_error_sync_main.js' },
+    { name: 'message/async_error_sync_esm.mjs' },
+    { name: 'message/error_aggregateTwoErrors.js', transform: errTransform },
+    { name: 'message/error_exit.js', transform: errTransform },
+    { name: 'message/error_with_nul.js', transform: errTransform },
+    { name: 'message/events_unhandled_error_common_trace.js', transform: errTransform },
+    { name: 'message/events_unhandled_error_nexttick.js', transform: errTransform },
+    { name: 'message/events_unhandled_error_sameline.js', transform: errTransform },
+    { name: 'message/events_unhandled_error_subclass.js', transform: errTransform },
+    { name: 'message/throw_custom_error.js', transform: errTransform },
+    { name: 'message/throw_error_with_getter_throw.js', transform: errTransform },
+    { name: 'message/throw_in_line_with_tabs.js', transform: errTransform },
+    { name: 'message/throw_non_error.js', transform: errTransform },
+    { name: 'message/throw_null.js', transform: errTransform },
+    { name: 'message/throw_undefined.js', transform: errTransform },
+    { name: 'message/promise_always_throw_unhandled.js', transform: promiseTransform },
+    { name: 'message/promise_unhandled_warn_with_error.js', transform: promiseTransform },
+    { name: 'message/assert_throws_stack.js', transform: promiseTransform },
+  ];
+  for (const { name, transform } of tests) {
+    it(name, async () => {
+      await snapshot.spawnAndAssert(fixtures.path(name), transform ?? defaultTransform);
+    });
+  }
+});
+
+describe('sourcemaps', { concurrency: true }, () => {
+  function normalize(str) {
+    return str.replaceAll(process.cwd(), '')
+    .replaceAll('//', '*')
+    .replaceAll('/Users/bencoe/oss/coffee-script-test', '')
+    .replaceAll(/\/(\w)/g, '*$1')
+    .replaceAll('*test*', '*')
+    .replaceAll('*fixtures*source-map*', '*');
+  }
+  const defaultTransform = snapshot.transform(snapshot.replaceWindowsLineEndings, replaceNodeVersion, normalize);
+
+  const tests = [
+    { name: 'message/source_map_disabled_by_api.js' },
+    { name: 'message/source_map_enabled_by_api.js' },
+    { name: 'message/source_map_enclosing_function.js' },
+    { name: 'message/source_map_eval.js' },
+    { name: 'message/source_map_no_source_file.js' },
+    { name: 'message/source_map_reference_error_tabs.js' },
+    { name: 'message/source_map_sourcemapping_url_string.js' },
+    { name: 'message/source_map_throw_catch.js' },
+    { name: 'message/source_map_throw_first_tick.js' },
+    { name: 'message/source_map_throw_icu.js' },
+    { name: 'message/source_map_throw_set_immediate.js' },
+  ];
+  for (const { name, transform } of tests) {
+    it(name, async () => {
+      await snapshot.spawnAndAssert(fixtures.path(name), transform ?? defaultTransform);
+    });
+  }
+});
+
+describe('vm', { concurrency: true }, () => {
```

Comment:
My guess is that much earlier in the project, the subfolders under `test/` corresponded with _how_ the test was run: `message` (`tools/test.py` compared output), `parallel` (the tests ran in parallel) etc. But nowadays everything runs in parallel and we’re continuing to move logic out of `test.py` into the tests themselves, and many subfolders of `test/` have to do with subsystems like `test/es-module`. This PR could be an opportunity to move all these former `test/message` tests into folders for their subsystems, like everything in this block could go under `test/vm`. Yes we could also do it in a follow-up but since this PR already moves dozens of files around, I kind of feel like we should decide how we want to organize the `test/` folder going forward and just do it, rather than delaying.

Reply:
+1 on reorganizing it, don't think it should be part of this PR

Reply:
Can we at least split up the new huge `test-node-output.mjs` file into one each for each of the `describe` blocks you have there? Since each `describe` block is for a separate subsystem. Then the new files could cleanly move into the new folders when we reorganize.

Likewise, rather than create a new `fixtures/message`, can you create `fixtures/async-error`, `fixtures/vm` etc to put the new files in? We don’t need to reorganize the world in this PR, but the files we’re creating or moving could at least go into folders divided by subsystem.

Reply:
👍🏻 perhaps I will break this up into multiple PRs

Reply:
If you want, though I think it can all go into one. This PR also creates the new snapshot generation logic so if you split it up, you’d need a primary one that lands first with that code and then follow-ups with each batch of tests migrated. Feels like a lot of work when I think everyone is fine with landing this all as one.